provider/pkg/schema/gen.go

```diff
@@ -1210,37 +1212,70 @@ func (ctx *context) propertyTypeSpec(parentName string, propSchema *jsschema.Sch
 		}, nil
 	}
 
+	if propSchema.Items != nil && len(propSchema.Items.Schemas) > 0 {
+		elementType, err := ctx.propertyTypeSpec(parentName+"Item", propSchema.Items.Schemas[0])
+		if err != nil {
+			return nil, err
+		}
+		return &pschema.TypeSpec{
+			Type:  "array",
+			Items: elementType,
+		}, nil
+	}
+
+	if propSchema.Items != nil {
```

Comment:
just checking for `Items` seems like it's throwing away the work you did in `NormalizeTypes` above. And we'd mistype the following schema: 
``` json
{
   "type": ["string", "array"]
   "items": { "type": "string" }
} 
```
as `[]string` when it should evaluate to `Union<string, []string>`

And in theory, we also have this problem with types that have an `object` in the type list too.

Instead, I think this logic needs to be nested in the evaluation of the list of types.

Reply:
Yes, this does generalise `Union<string, []string>` to `[]string` currently which isn't quite as accurate technically, but is probably simpler and therefore more usable in some languages (e.g. Go) where unions are not as well supported.

Reply:
Now that we have `NormalizeTypes`, why not use it for `Items.Schemas` as well? Wouldn't fix the issues Matt pointed out?